lib/compute/jenkins-main-node.ts

```diff
@@ -288,8 +298,40 @@ export class JenkinsMainNode {
+      InitCommand.shellCommand(`java -jar /jenkins-cli.jar -s http://localhost:8080 -auth @/var/lib/jenkins/secrets/myIdPassDefault install-plugin ${JenkinsPlugins.plugins.join(' ')} && java -jar jenkins-cli.jar -s http://localhost:8080 -auth @/var/lib/jenkins/secrets/myIdPassDefault restart`),
       // Warning : any commands after this may be executed before the above command is complete
+
+      // Commands are fired one after the other but it does not wait for the command to complete.
+      // Therefore, sleep 60 seconds to wait for plugins to install and jenkins to start which is required for the next step
+      InitCommand.shellCommand('sleep 60'),
+
+      // If devMode is false, first line extracts the oidcFederateProps as json from the secret manager
+      // xmlstarlet is used to setup the securityRealm values for oidc
+      InitCommand.shellCommand(oidcFederateProps.devMode ? 'echo Not altering the jenkins config.xml in dev-mode'
+        // eslint-disable-next-line max-len
+        : `var=\`aws --region ${stackRegion} secretsmanager get-secret-value --secret-id ${oidcFederateProps.oidcCredArn} --query SecretString --output text\` &&`
+        + 'xmlstarlet ed -L -d "/hudson/securityRealm" \\'
+        + '-s /hudson -t elem -n securityRealm -v " " \\'
+        + '-i //securityRealm -t attr -n "class" -v "org.jenkinsci.plugins.oic.OicSecurityRealm" \\'
+        + '-i //securityRealm -t attr -n "plugin" -v "oic-auth@1.8" \\'
+        + '-s /hudson/securityRealm -t elem -n clientId -v "$(echo $var | jq -r ".clientId")" \\'
+        + '-s /hudson/securityRealm -t elem -n clientSecret -v "$(echo $var | jq -r ".clientSecret")" \\'
+        + '-s /hudson/securityRealm -t elem -n wellKnownOpenIDConfigurationUrl -v "$(echo $var | jq -r ".wellKnownOpenIDConfigurationUrl")" \\'
+        + '-s /hudson/securityRealm -t elem -n tokenServerUrl -v "https://idp-integ.federate.amazon.com/api/oauth2/v2/token" \\'
+        + '-s /hudson/securityRealm -t elem -n authorizationServerUrl -v "https://idp-integ.federate.amazon.com/api/oauth2/v1/authorize" \\'
+        + '-s /hudson/securityRealm -t elem -n userInfoServerUrl -v "https://idp-integ.federate.amazon.com/api/oauth2/v1/userinfo" \\'
+        + '-s /hudson/securityRealm -t elem -n userNameField -v "sub" \\'
+        + '-s /hudson/securityRealm -t elem -n scopes -v "openid" \\'
+        + '-s /hudson/securityRealm -t elem -n disableSslVerification -v false \\'
+        + '-s /hudson/securityRealm -t elem -n logoutFromOpenidProvider -v "true" \\'
+        + '-s /hudson/securityRealm -t elem -n postLogoutRedirectUrl -v "" \\'
+        + '-s /hudson/securityRealm -t elem -n escapeHatchEnabled -v "false" \\'
+        + '-s /hudson/securityRealm -t elem -n escapeHatchSecret -v "dsafsdfasfasdf" \\'
```

Comment:
I am not sure at this point as to what value goes in here. I am looking into this. It's a dummy value for now.